homework/ba-nguyen/lesson-10/js-5.md

```diff
@@ -0,0 +1,274 @@
+
+    for (let i = 0; i < tabLength; i++) {
+      let tabButton = document.createElement("button");
+
+      tabButton.textContent = document
+        .querySelectorAll("div")
+        [i].getAttribute("data-tabname");
+      // By default, hide the body of the tabs.
+      node.querySelectorAll("div")[i].classList.add("hidden");
+      tabButton.addEventListener("click", switchTab);
+      node.insertBefore(tabButton, document.querySelector("div"));
+    }
+    // Sets the initially selected tab & display its associated content.
+    activeTab = document.querySelector("button");
+    activeTab.classList.add("selected");
+    currentBody = document.querySelector("div");
+    currentBody.classList.toggle("hidden");
+  }
+
+  asTabs(document.querySelector("tab-panel"));
+</script>
+```
+
+**Question 4:**
+
+```css
+input[type="range"] {
+  accent-color: #f44336;
+}
+```
+
+**Question 5: What's the output?**
+
+```js
+const myPromise = Promise.resolve(Promise.resolve("Promise"));
+
+function funcOne() {
+  setTimeout(() => console.log("Timeout 1!"), 0);
+  myPromise.then((res) => res).then((res) => console.log(`${res} 1!`));
+  console.log("Last line 1!");
+}
+
+async function funcTwo() {
+  const res = await myPromise;
+  console.log(`${res} 2!`);
+  setTimeout(() => console.log("Timeout 2!"), 0);
+  console.log("Last line 2!");
+}
+
+funcOne();
+funcTwo();
+```
+
+- A: Promise 1! Last line 1! Promise 2! Last line 2! Timeout 1! Timeout 2!
+- B: Last line 1! Timeout 1! Promise 1! Last line 2! Promise2! Timeout 2!
+- `C: Last line 1! Promise 2! Last line 2! Promise 1! Timeout 1! Timeout 2!` => **Answer**
+- D: Timeout 1! Promise 1! Last line 1! Promise 2! Timeout 2! Last line 2!
+
+**Question 6: How can we invoke sum in sum.js from index.js?**
+
+```js
+// sum.js
+export default function sum(x) {
+  return x + x;
+}
+
+// index.js
+import * as sum from "./sum";
+```
+
+- A: sum(4)
+- B: sum.sum(4)
+- `C: sum.default(4)` => **Answer**
+- D: Default aren't imported with \*, only named exports
+
+**Question 7: What's the output?**
+
+```js
+const handler = {
+  set: () => console.log("Added a new property!"),
+  get: () => console.log("Accessed a property!"),
+};
+
+const person = new Proxy({}, handler);
+
+person.name = "Lydia";
+person.name;
+```
+
+- `A: Added a new property!` => **Answer**
+- B: Accessed a property!
+- C: Added a new property! Accessed a property!
+- D: Nothing gets logged
+
+**Question 8: Which of the following will modify the person object?**
+
+```js
+const person = { name: "Lydia Hallie" };
+
+Object.seal(person);
+```
+
+- `A: person.name = "Evan Bacon"` => **Answer**
+- B: person.age = 21
+- C: delete person.name
+- D: Object.assign(person, { age: 21 })
+
+> Explain: The method **seal** of the Object is mean that you can't modify the existing properties and you can't add new properties
+
+**Question 9: What's the output?**
+
+```js
+const add = (x) => x + x;
+
+function myFunc(num = 2, value = add(num)) {
```

Comment:
thaycacac: new thing to me, can you give the keyword so i can continue researching?

Reply:
just javascript runtime